odoo/addons/base/tests/test_float.py

```diff
@@ -82,6 +84,34 @@ def try_round(amount, expected, digits=3, method='HALF-UP'):
         try_round(457.4554, '457.455')
         try_round(-457.4554, '-457.455')
 
+        # Try some rounding value with rounding method HALF-DOWN instead of HALF-UP
+        try_round(2.6735, '2.673', method='HALF-DOWN')  # Tie rounds towards 0
+        try_round(-2.6735, '-2.673', method='HALF-DOWN')  # Tie rounds towards 0
+        try_round(2.6745, '2.674', method='HALF-DOWN')  # Tie rounds towards 0
+        try_round(-2.6745, '-2.674', method='HALF-DOWN')  # Tie rounds towards 0
+        try_round(2.6744, '2.674', method='HALF-DOWN')
+        try_round(-2.6744, '-2.674', method='HALF-DOWN')
+        try_round(0.0004, '0.000', method='HALF-DOWN')
+        try_round(-0.0004, '-0.000', method='HALF-DOWN')
+        try_round(357.4555, '357.455', method='HALF-DOWN')
+        try_round(-357.4555, '-357.455', method='HALF-DOWN')
+        try_round(457.4554, '457.455', method='HALF-DOWN')
+        try_round(-457.4554, '-457.455', method='HALF-DOWN')
+
+        # Try some rounding value with rounding method HALF-EVEN instead of HALF-UP
+        try_round(2.6735, '2.674', method='HALF-EVEN')  # Tie rounds to the closest even number (i.e. up here)
+        try_round(-2.6735, '-2.674', method='HALF-EVEN')  # Tie rounds to the closest even number (i.e. up here)
+        try_round(2.6745, '2.674', method='HALF-EVEN')  # Tie rounds to the closest even number (i.e. down here)
+        try_round(-2.6745, '-2.674', method='HALF-EVEN')  # Tie rounds to the closest even number (i.e. down here)
+        try_round(2.6744, '2.674', method='HALF-EVEN')
+        try_round(-2.6744, '-2.674', method='HALF-EVEN')
+        try_round(0.0004, '0.000', method='HALF-EVEN')
+        try_round(-0.0004, '-0.000', method='HALF-EVEN')
+        try_round(357.4555, '357.455', method='HALF-EVEN')
+        try_round(-357.4555, '-357.455', method='HALF-EVEN')
+        try_round(457.4554, '457.455', method='HALF-EVEN')
```

Comment:
457.555 doesn't seem very even to me, so not sure why you'd expect it when rounding to the nearest even number.
Noticed this working on https://github.com/odoo/odoo/pull/152709 which inadvertently fixes the issue.

Reply:
Indeed these two values were wrong :grimacing: